assignments/minishell/README.md

```diff
@@ -0,0 +1,164 @@
+
+### Changing the current directory
+
+The shell will have a `cd` built-in command that will provide the user a way to navigate the file system.
+To implement this feature you will need to store the current directory path because the user can provide either relative or absolute paths to the `cd` command.
+
+The built-in `pwd` command will show the current directory path.
+
+Follow the following examples to understand how the functionality should behave.
+
+```sh
+> pwd
+/home/student
+> cd operating-systems/assignments/minishell
+> pwd
+/home/student/operating-systems/assignments/minishell
+> cd inexitent
+no such file or directory
+> cd /usr/lib
+> pwd
+/usr/lib
+```
+
+### Closing the shell
+
+Inputting either `quit` or `exit` commands should lead to closing the minishell.
+
+### Environment variables
+
+Your shell will support using environment variables.
+The environment variables will be initially inherited from the `shell` process that opened your minishell application.
+
+If an undefined variable is used, it is supposed that the variable has the empty string value `""`.
+
+```sh
+> NAME="John Doe"                    # Will assign "John Doe" to NAME variable
+> AGE=27                             # Will assign 27 to AGE variable
+> ./identify $NAME $LOCATION $AGE    # Will translate to ./identify "John Doe" "" 27 because $LOCATION is not defined
+```
+
+A variable can be assigned to another variable.
+
+```sh
+> OLD_NAME=$NAME    # Will assign the value of the NAME variable to OLD_NAME
+```
+
+### Running an application
+
+Suppose you have an executable named `sum` in the current directory that takes as many numbers as arguments and sums them, printing the result at `stdout`.
+The following example shows how the minishell implemented by you should behave.
+
+```sh
+> ./sum 2 4 1
+7
+```
+
+If the executable is located at the `/home/student/sum` absolute path, the following example should also be valid.
+
+```sh
+> /home/student/sum 2 4 1
+7
+```
+
+Each application will run in a separate child process of the minishell created using [fork](https://man7.org/linux/man-pages/man2/fork.2.html).
+
+### Operators
+
+#### Sequential operator
+
+Using the `;` operator you can chain multiple commands that will run sequentially, one after another.
+In the command `expr1; expr2` it is guaranteed that `expr1` evaluation will finish before `expr2` will be evaluated.
+
+```sh
+> echo "Hello"; echo "world!"; echo "Bye!"
+Hello
+world!
+Bye!
+```
+
+#### Parallel operator
+
+Using the `&` operator you can chain multiple commands that will run in parallel.
+When running the command `expr1 & expr2`, the evaluation for both expressions start at the same time, and it is not guaranteed to finish in a specific order.
+
+```sh
+> echo "Hello" & echo "world!" & echo "Bye!"
+world!
+Hello
+Bye!
+```
+
+#### Pipe operator
+
+Using the `|` operator you can chain multiple commands that will pass the output of the previous command as the input for the current command.
```

Comment:
```suggestion
With the `|` operator you can chain multiple commands so that the standard output of the first command is redirected to the standard input of the second command.
```